sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -119,7 +120,7 @@ object Project {
       case (StructType(fields), expected: StructType) =>
         val newFields = reorderFields(
           fields.zipWithIndex.map { case (f, index) =>
-            (f.name, GetStructField(col, index))
+            (f.name, GetStructField(AssertNotNull(col, columnPath), index))
```

Comment:
shall we use `KnownNotNull`?

Reply:
and we should only add it if `col.nullable`

Reply:
Oh, there is `KnownNotNull`, good to know.
Sure, will submit a followup.

Reply:
#40546